### PR TITLE
Add chmod/chmodSync on *nix (and fix Cargo.toml)

### PR DESCRIPTION
**Pull request by kevinkassimo**
Continuation from #673 (since no updates for quite a while and this is an important fs call)

Implements `chmod` and `chmodSync` on *nix. (On windows this is noop)

Prior art:
+ Go: `os.Chmod(name string, mode FileMode) error`
+ Node: `fs.chmod(file, mode, callback);`
+ Python: `os.chmod(path, mode);`
+ Rust: `std::fs::set_permissions<P: AsRef>(path: P, perm: Permissions) -> Result<()>`

`lchmod` is not implemented yet. It seems that Rust `std::fs` only support inspecting symlink metadata, but any `set_permissions/set_mode` calls would eventually set the permission of linked file/dir.

Also, add `getopts` entry in `Cargo.toml` (forgot to include this in the PR yesterday)

### CODE REVIEW

File: js/chmod_test.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    err = e;
+  }
+  assertEqual(err.kind, deno.ErrorKind.NotFound);
+  assertEqual(err.name, "NotFound");
+});
+
+testPerm({ write: false }, function chmodSyncPerm() {
+  let err;
+  try {
+    const filename = deno.makeTempDirSync() + "/test.txt";
```
Comment (J2P) [inline js/chmod_test.ts]: I think `deno.makeTempDirSync()` raise a PermissionDenied error.

Comment (kevinkassimo) [reply js/chmod_test.ts]: Ooops. Thanks for pointing out

**Comment by ry** [comment]
On windows there are a bunch of errors like this
```
C:/deno/js/chmod_test.ts:18:8 - error TS2339: Property 'chmodSync' does not exist on type 'typeof import("deno")'.
18   deno.chmodSync(filename, 0o777);
          ~~~~~~~~~
```
Otherwise looks good!

**Comment by ry** [comment]
I suspect the Windows problem is similar to what we saw in the CheckJS https://github.com/denoland/deno/pull/1090 patches - where Windows is not properly detecting changes to the bundle (maybe when a new file is added?) There were changes to this mechanism in fd68f85ce... maybe I introduced a bug.

**Comment by srijanreddy98** [comment]
@ry @kevinkassimo can you suggest me what to do next.

**Pull request merged**